front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/adjustConfWithTrainToModify.ts

```diff
@@ -49,46 +42,60 @@ export function loadPathFinding(path: PathResponse, dispatch: Dispatch) {
           (via, idx) => idx !== 0 && path.steps && idx < path.steps.length - 1 && !via.suggestion
         )
         .map((via) => convertStepToPointOnMap(via));
-      dispatch(replaceVias(compact(vias)));
-      dispatch(updateSuggeredVias(compact(path.steps.map((via) => convertStepToPointOnMap(via)))));
+      dispatch(osrdActions.replaceVias(compact(vias)));
+      dispatch(
+        osrdActions.updateSuggeredVias(
+          compact(path.steps.map((via) => convertStepToPointOnMap(via)))
+        )
+      );
     } else {
-      dispatch(replaceVias([]));
+      dispatch(osrdActions.replaceVias([]));
     }
   }
 }
 
 export default function adjustConfWithTrainToModify(
   trainSchedule: TrainSchedule,
   path: PathResponse,
-  dispatch: Dispatch
+  dispatch: Dispatch,
+  usingElectricalProfiles: boolean,
+  osrdActions: OperationalStudiesConfSliceActions | StdcmConfSliceActions
 ) {
-  const { osrdconf } = store.getState();
+  const {
+    updateRollingStockID,
+    toggleUsingElectricalProfiles,
+    updateSpeedLimitByTag,
+    updateLabels,
+    updateName,
+    updateDepartureTime,
+    updateInitialSpeed,
+    updateAllowances,
+  } = osrdActions;
   if (trainSchedule.rolling_stock_id)
     dispatch(updateRollingStockID(trainSchedule.rolling_stock_id));
-  if (
-    trainSchedule.options?.ignore_electrical_profiles ===
-    osrdconf.simulationConf.usingElectricalProfiles
-  )
+
+  if (trainSchedule.options?.ignore_electrical_profiles === usingElectricalProfiles)
     dispatch(toggleUsingElectricalProfiles());
 
   dispatch(updateSpeedLimitByTag(trainSchedule.speed_limit_tags));
   dispatch(updateLabels(trainSchedule.labels));
 
   if (trainSchedule.train_name) dispatch(updateName(trainSchedule.train_name));
+
   if (trainSchedule.departure_time)
     dispatch(updateDepartureTime(sec2time(trainSchedule.departure_time)));
   dispatch(updateInitialSpeed(ms2kmh(trainSchedule.initial_speed) || 0));
 
   if (trainSchedule.allowances) dispatch(updateAllowances(trainSchedule.allowances as Allowance[]));
 
-  loadPathFinding(path, dispatch);
+  loadPathFinding(path, dispatch, osrdActions);
 
-  if (trainSchedule.allowances) dispatch(updateAllowances(trainSchedule.allowances));
+  if (trainSchedule.allowances) dispatch(osrdActions.updateAllowances(trainSchedule.allowances));
 
   if (trainSchedule.power_restriction_ranges) {
     // TODO: refacto this when using the new train_schedule endpoints
     dispatch(
-      updatePowerRestrictionRanges(
+      osrdActions.updatePowerRestrictionRanges(
```

Comment:
destructure updatePowerRestrictionRanges l73
```suggestion
      updatePowerRestrictionRanges(
```

Reply:
Done